### ISSUE TITLE
MinMax Normalization

### ISSUE THREAD

**Issue by BananaLoaf** [keras-team/keras#15448]
**System information**.

TensorFlow version (you are using): **2.6.0**
Are you willing to contribute it (Yes/No): **Yes**

**Describe the feature and the current behavior/state**.

Current documentation states that to normalize values [0, 255] between [-1, 1] we need to use Rescaling layer with `scale=1./127.5, offset=-1`

Rescaling layer math looks like `f(x) = x * scale + offset` where:

`scale = 1/ ((max - min) / (new_max - new_min))`
`scale = (new_max - new_min) / (max - min)`

`offset=new_min`

Therefore:
`f(x) = x * (new_max - new_min) / (max - min) + new_min`

It works perfectly for `x ∈ [min, max], min = 0, max > 0`, however for `x ∈ [min, max], min ≠ 0` transformation fails, because `min` shift is not accounted for:
`f(x) = (x - min) * (new_max - new_min) / (max - min) + new_min`, which looks exactly like MinMaxNormalization, hence the proposal.

Current API requires users to shift values beforehand so that `min = 0` or doing something like this:
`keras.layers.Rescaling(scale=(new_max-new_min)/(max - min), offset=new_min)(x-min)` with `min` defined where ever, which I'm not sure it won't break back-propagation.
_or_
`tf.keras.layers.Lambda(lambda x: (x - min) * (new_max - new_min) / (max - min) + new_min)(x)`, and to quote [Lambda source code](https://github.com/keras-team/keras/blob/v2.6.0/keras/layers/core.py#L741-L1063): 

> WARNING: `tf.keras.layers.Lambda` layers have (de)serialization limitations!

**Will this change the current api? How?**
Currently and IMO Rescaler layers feels very limited, hence I propose to add a new MinMaxNormalization layer

**Who will benefit from this feature?**

Whomever needs to normalize data that has minimum values not equal to 0 (i.e Celsius temperature, wind speed vector, turning image from [-1, 1] back into [0, 255])

- Do you want to contribute a PR? (yes/no): **Yes**
- Briefly describe your candidate solution(if contributing): I will add a new _experimental_ layer
